### PR TITLE
feat: create api to nominate vote

### PR DESCRIPTION
**Pull request by LongTCH**

### CODE REVIEW

File: app/services/room.js [hunk 1]
```diff
@@ -1,9 +1,39 @@
...
+      } else {
+        // if not, that mean user cancelled a vote
+        // we need to set 'vote' for this user is null
+        userVoting.vote = null;
+      }
+      voting[userVotingIndex] = userVoting;
+      await room.save();
+      return true;
+    } catch {
+      return false;
```
Comment (ces-hoangdao) [inline app/services/room.js:36]: we don't have to try catch here, just throw the error and put the try catch block in the controller

File: app/services/message.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const RESPONSE_MESSAGE = {
```
Comment (ces-hoangdao) [inline app/services/message.js:1]: this should be put in the constants folder

File: app/services/room.js [hunk 3]
```diff
@@ -1,9 +1,39 @@
...
+
+  async nominateVote({ roomId, userId, vote, connected }) {
+    try {
+      const room = await Room.findById(roomId);
+      const voting = room.voting;
+      const userVotingIndex = voting.findIndex(
+        (_userVoting) => _userVoting.user == userId
+      );
+      // if userId not found, that mean that user hasn't join room
+      if (userVotingIndex == -1) return false;
```
Comment (ces-hoangdao) [inline app/services/room.js:18]: ===, should check for the type too

Comment (ces-hoangdao) [inline app/services/room.js:18]: Create a proper error message for this, this should be some kind like "Unable to find the user"

Comment (LongTCH) [reply app/services/room.js:18]: I don't know, but we store 'user' in 'voting' as ObjectId and request send 'userId' as String, so I want to compare the values between them.

File: app/services/message.js (hunk 2 above)
Comment (LongTCH) [reply app/services/message.js:1]: yes, that was my mistake.

File: app/services/room.js (hunk 1 above)
Comment (LongTCH) [reply app/services/room.js:36]: I got it

File: app/services/room.js (hunk 3 above)
Comment (ces-hoangdao) [reply app/services/room.js:18]: @LongTCH  for [FYI](https://www.freecodecamp.org/news/loose-vs-strict-equality-in-javascript/#:~:text=The%20%3D%3D%20operator%20performs%20a,well%20as%20the%20same%20value).)

Comment (LongTCH) [reply app/services/room.js:18]: yes

Comment (LongTCH) [reply app/services/room.js]: OK

**Review by ces-hoangdao — APPROVED** [review]

**Pull request merged**